component/src/main/ballerina/cassandra/endpoint.bal

```diff
@@ -0,0 +1,230 @@
+    int refreshNodeIntervalMillis = -1,
+    int refreshSchemaIntervalMillis = -1,
+};
+
+
+documentation {
+    Options related to connection pooling.
+
+    F{{maxRequestsPerConnectionLocal}} The maximum number of requests per connection for local hosts
+    F{{maxRequestsPerConnectionRemote}} The maximum number of requests per connection for remote hosts
+    F{{idleTimeoutSeconds}} The timeout before an idle connection is removed
+    F{{poolTimeoutMillis}} The timeout when trying to acquire a connection from a host's pool
+    F{{maxQueueSize}} The maximum number of requests that get enqueued if no connection is available
+    F{{heartbeatIntervalSeconds}} The heart beat interval, after which a message is sent on an idle connection
+    to make sure it's still alive
+    F{{coreConnectionsPerHostLocal}} The core number of connections per local host
+    F{{maxConnectionsPerHostLocal}} The maximum number of connections per local host
+    F{{newConnectionThresholdLocal}} The threshold that triggers the creation of a new connection to a local host
+    F{{coreConnectionsPerHostRemote}} The core number of connections per remote host
+    F{{maxConnectionsPerHostRemote}} The maximum number of connections per remote host
+    F{{newConnectionThresholdRemote}} The threshold that triggers the creation of a new connection to a remote host
+}
+public type PoolingOptionsConfiguration {
+    int maxRequestsPerConnectionLocal = -1,
+    int maxRequestsPerConnectionRemote = -1,
+    int idleTimeoutSeconds = -1,
+    int poolTimeoutMillis = -1,
+    int maxQueueSize = -1,
+    int heartbeatIntervalSeconds = -1,
+    int coreConnectionsPerHostLocal = -1,
+    int maxConnectionsPerHostLocal = -1,
+    int newConnectionThresholdLocal = -1,
+    int coreConnectionsPerHostRemote = -1,
+    int maxConnectionsPerHostRemote = -1,
+    int newConnectionThresholdRemote = -1,
+};
+
+documentation {
+    Options to configure low-level socket options for the connections kept to the Cassandra hosts.
+
+    F{{connectTimeoutMillis}} The connection timeout in milliseconds
+    F{{readTimeoutMillis}} The per-host read timeout in milliseconds
+    F{{soLinger}} The linger-on-close timeout
+    F{{receiveBufferSize}} A hint to the size of the underlying buffers for incoming network I/O
+    F{{sendBufferSize}} A hint to the size of the underlying buffers for outgoing network I/O
+}
+public type SocketOptionsConfiguration {
+    int connectTimeoutMillis = -1,
+    int readTimeoutMillis = -1,
+    int soLinger = -1,
+    int receiveBufferSize = -1,
+    int sendBufferSize = -1,
+
+// TODO: Driver do not set these by default. It takes the default values of the underlying netty transport.
+// But if we take the inputs as booleans and if the value set in the bal file is "false", we wouldn't know
+// if it was set by the user or not. So need to decide how to handle this.
+// boolean keepAlive;
+// boolean reuseAddress;
+// boolean tcpNoDelay;
```

Comment:
Shall we create an issue for this and remove the commented lines

Reply:
There is an issue already. We can just remove the TODO.